src/client/retry.rs

```diff
@@ -32,7 +32,7 @@ use tracing::info;
 use web_time::{Duration, Instant};
 
 /// Retry request error
-#[derive(Debug, thiserror::Error)]
+#[derive(Debug)]
```

Comment:
this is technically a breaking API change, I think

However the next release appears to be slated to be a breaking one so that would be ok: https://github.com/apache/arrow-rs?tab=readme-ov-file#release-versioning-and-schedule

We could potentially put in a workaround like implement a non standard way to access the source if we want to put it in before the next breaking change

Reply:
I disagree because:

1. this type isn't public, hence it can only be used as via the trait `dyn Error`. You CANNOT downcast it.
2. All this derive does is to implement `std::error::Error`. Also see [thiserror docs](https://docs.rs/thiserror/latest/thiserror/#details) that state "Thiserror deliberately does not appear in your public API." I.e. removing this is a change of an implementation detail, NOT a public API.

Reply:
Makes sense to me -- thank you for the clarification